### DISCUSSION TITLE
Custom Commands aren't Triggering

### DISCUSSION THREAD

**Discussion by debugsoul** [GrapesJS/grapesjs#5866] [Q&A]
Upon delving into the intricacies of GrapesJS within the Angular framework, I have distilled my immediate needs into two distinct objectives, each representing a unique facet of my journey into this realm of web development:

1. Custom Button Creation and Event Handling:
 ![image](https://github.com/GrapesJS/grapesjs/assets/42071761/f68adf05-5a69-475b-bcac-974894618b7f)
2. Creation of a New Blank Container (View):
![image](https://github.com/GrapesJS/grapesjs/assets/42071761/15016d7b-8f07-404f-8dd0-48ab61288146)

My Current Code:
` loadNewsLetter(): void {
    let host = 'https://grapesjs.com/';
    this.editor = grapesjs.init({
      selectorManager: { componentFirst: true },
      clearOnRender: true,
      commands: {
        strict: false, // by default is true
      },
      container: '#gjs',
      fromElement: true,
     width: 'auto',
      panels: { defaults: [] },
      plugins: [newsLetter, ckeditor, exportFile,grapejsEvent],
      pluginsOpts: {
        [exportFile]: { addExportBtn: true },
        [newsLetter]: {  modalLabelImport: 'Paste all your code here below and click import',
        modalLabelExport: 'Copy the code and use it wherever you want',
        codeViewerTheme: 'material',
        importPlaceholder: '<table class="table"><tr><td class="cell">Hello world!</td></tr></table>',
        cellStyle: {
          'font-size': '12px',
          'font-weight': 300,
          'vertical-align': 'top',
          color: 'rgb(111, 119, 125)',
          margin: 0,
          padding: 0,
        },
        [exportFile]: { onToolbar: (el: { style: { minWidth: string; }; }) => {
          el.style.minWidth = '350px';
        },
        options: {
          startupFocus: true,
          extraAllowedContent: '*(*);*{*}', // Allows any class and any inline style
          allowedContent: true, // Disable auto-formatting, class removing, etc.
          enterMode: 2, // CKEDITOR.ENTER_BR,
          extraPlugins: 'sharedspace,justify,colorbutton,panelbutton,font',
          toolbar: [
            { name: 'styles', items: ['Font', 'FontSize' ] },
            ['Bold', 'Italic', 'Underline', 'Strike'],
            {name: 'paragraph', items : [ 'NumberedList', 'BulletedList']},
            {name: 'links', items: ['Link', 'Unlink']},
            {name: 'colors', items: [ 'TextColor', 'BGColor' ]},
          ],
        } }}
      }
    });
    this.cleanPanel();
  }`

Additional Button:
`  cleanPanel(): void {

      this.editor.Panels.addButton('options',
 [{
   id: 'save-db',
   className: 'fas fa-save',
   command: 'save-db',
   attributes: {
     title: 'Save Changes'
   }
 }]
);

this.editor.Commands.add('save-db', {
  run: function(editor, sender) {
    sender && sender.set('active', 0); // turn off the button
    editor.store();
    //storing values to variables
    var htmldata = editor.getHtml();
    var cssdata = editor.getCss();
    alert('aa');
  }
 });

  }`

**Comment by artf** [comment]
Please provide a reproducible demo

**Reply by debugsoul** [reply]
Thanks for the your response, I figure it out the issue, I think it because of font awesome class whenever tried to create the font awesome Icon it not generating the span wrapper classname with gjs-pn-btn.

But whenever we tried our custom css it's worked just fine took a while to figure but I believe that was the issue only with label it's working fine. 

` cleanPanel(): void {
    this.editor.Panels.addButton('options',
    [{
        id: 'save-db',
        name: 'save-db',
        className: 'fa fa-save',  --this is creating the problem. 
        //attributes: {class: 'fa fa-save', title: "Edit Menu"},
        //label: 'Save',
        command: (editor,sender) => {
          sender && sender.set('active', 0); // turn off the button
          editor.store();
          var html = editor.runCommand('gjs-get-inlined-html');
          var htmldata = editor.getHtml();
          var cssdata = editor.getCss();
          console.log(htmldata);
          console.log(cssdata);
          console.log(html);
        },
        attributes: {
          title: 'Save Changes'
        }
    }]);
  }`

![image](https://github.com/GrapesJS/grapesjs/assets/42071761/e08f94a9-c8e2-43cd-8a47-ffe3431d87c1)
